ARMmbed/mbedtls
# Support building on e2k (Elbrus) architecture

Signed-off-by: makise-homura <akemi_homura@kurisa.ch>

Edited on: 08/22/2020 10:34 PM UTC

## Description
This pull request introduces support for building mbedtls on Elbrus hardware platform (which is based on Russian [Elbrus](https://en.wikipedia.org/wiki/Elbrus-8S) CPU family) with its native lcc (eLbrus Compiler Collection) compiler.

## Status
**READY**

## Requires Backporting
~~NO (This PR is a new feature\enhancement)~~ Ended up in YES, done: PR #3598 and #3599 .

## Migrations
NO

## Additional comments
This PR correctly enables `-Wformat-signedness` (which may or may not be supported by lcc), and ~~pass `-Wno-signed-one-bit-field` and `-Wno-sign-compare` (if they supported) to get rid of excess warnings~~ fixes issues that caused `-Wsigned-one-bit-field` and `-Wsign-compare` warnings, and also fixes a suspicious point in `dh_genprime`.

## Todos

## Steps to test or reproduce
To reproduce, you should have an Elbrus workstation with OS Elbrus >= 5.0-rc2 and CPU which is Elbrus-8C (Elbrus-8S) or newer (you may ask contributor for SSH access to demo station).
Building with `mkdir build; cd build; cmake .. && make -j16` should succeed and produce no warnings.
